datadog-profiling/src/internal/profile/otel_emitter/profile.rs

```diff
@@ -0,0 +1,939 @@
+                original_payload_format: String::new(), // There is no original payload
+                original_payload: vec![],    // There is no original payload
+                attribute_indices: vec![],   // There are currently no attributes at this level
+            };
+
+            profiles.push(profile);
+        }
+
+        for (sample, timestamp, mut values) in std::mem::take(&mut self.observations).into_iter() {
+            let stack_index = sample.stacktrace.to_raw_id() as i32;
+            let label_set = self.get_label_set(sample.labels)?;
+            let attribute_indicies: Vec<_> =
+                label_set.iter().map(|x| x.to_raw_id() as i32).collect();
+            let labels = label_set
+                .iter()
+                .map(|l| self.get_label(*l).copied())
+                .collect::<Result<Vec<_>>>()?;
+            let link_index = 0; // TODO, handle links properly
+            let timestamps_unix_nano = timestamp.map_or(vec![], |ts| vec![ts.get() as u64]);
+            self.upscaling_rules.upscale_values(&mut values, &labels)?;
+
+            for (idx, value) in values.iter().enumerate() {
+                if *value != 0 {
+                    let otel_sample = datadog_profiling_otel::Sample {
+                        stack_index,
+                        attribute_indices: attribute_indicies.clone(),
+                        link_index,
+                        values: vec![*value],
+                        timestamps_unix_nano: timestamps_unix_nano.clone(),
+                    };
+                    profiles[idx].sample.push(otel_sample);
+                }
+            }
+        }
+
+        // Convert string table using into_lending_iter
+        // Note: We can't use .map().collect() here because LendingIterator doesn't implement
+        // the standard Iterator trait. LendingIterator is designed for yielding references
+        // with lifetimes tied to the iterator itself, so we need to manually iterate and
+        // convert each string reference to an owned String.
+        let string_table = {
+            let mut strings = Vec::with_capacity(self.strings.len());
+            let mut iter = self.strings.into_lending_iter();
+            while let Some(s) = iter.next() {
+                strings.push(s.to_string());
+            }
+            strings
+        };
+
+        // Convert labels to KeyValues for the attribute table
+        let mut key_to_unit_map = HashMap::new();
+        let mut attribute_table = Vec::with_capacity(self.labels.len());
+
+        for label in self.labels.iter() {
+            let key_value = convert_label_to_key_value(label, &string_table, &mut key_to_unit_map)
+                .with_context(|| {
+                    format!(
+                        "Failed to convert label with key index {}",
+                        label.get_key().to_raw_id()
+                    )
+                })?;
+            attribute_table.push(key_value);
+        }
```

Comment:
Note that as I mentioned above, we're missing actually attaching the attributes to samples, so right now they're only a bag of unreferenced things here.